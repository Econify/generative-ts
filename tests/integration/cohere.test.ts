import {
  CohereChatApi,
  CohereGenerateApi,
  createCohereModelProvider,
} from "@packages/core";

test("Cohere - Generate", async () => {
  // arrange
  const cohereGen = createCohereModelProvider({
    api: CohereGenerateApi,
    modelId: "command",
  });

  // act
  const response = await cohereGen.sendRequest({
    prompt: "Brief History of NY Mets:",
    return_likelihoods: "ALL",
  });

  // assert
  expect(response).toMatchApiSnapshot();
});

test("Cohere - Chat", async () => {
  // arrange
  const cohereChat = createCohereModelProvider({
    api: CohereChatApi,
    modelId: "command-r-plus",
  });

  // act
  const response = await cohereChat.sendRequest({
    prompt: "Brief History of NY Mets:",
    preamble: "Talk like Jafar from Aladdin",
  });

  // assert
  expect(response).toMatchApiSnapshot();
});

test("Cohere - Chat (Default)", async () => {
  // arrange
  const cohereChat = createCohereModelProvider({
    modelId: "command-r-plus",
  });

  // act
  const response = await cohereChat.sendRequest({
    prompt: "Brief History of NY Mets:",
    preamble: "Talk like Jafar from Aladdin",
  });

  // assert
  expect(response).toMatchApiSnapshot();
});
