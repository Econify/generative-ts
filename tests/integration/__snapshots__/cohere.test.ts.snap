// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohere - Chat (Default) 1`] = `
{
  "chat_history": [
    {
      "message": "string",
      "role": "string",
    },
  ],
  "finish_reason": "string",
  "generation_id": "string",
  "meta": {
    "api_version": {
      "version": "string",
    },
    "billed_units": {
      "input_tokens": "number",
      "output_tokens": "number",
    },
    "tokens": {
      "input_tokens": "number",
      "output_tokens": "number",
    },
  },
  "response_id": "string",
  "text": "string",
}
`;

exports[`Cohere - Chat 1`] = `
{
  "chat_history": [
    {
      "message": "string",
      "role": "string",
    },
  ],
  "finish_reason": "string",
  "generation_id": "string",
  "meta": {
    "api_version": {
      "version": "string",
    },
    "billed_units": {
      "input_tokens": "number",
      "output_tokens": "number",
    },
    "tokens": {
      "input_tokens": "number",
      "output_tokens": "number",
    },
  },
  "response_id": "string",
  "text": "string",
}
`;

exports[`Cohere - Generate 1`] = `
{
  "generations": [
    {
      "finish_reason": "string",
      "id": "string",
      "likelihood": "number",
      "text": "string",
      "token_likelihoods": [
        {
          "likelihood": "number",
          "token": "string",
        },
      ],
    },
  ],
  "id": "string",
  "meta": {
    "api_version": {
      "version": "string",
    },
    "billed_units": {
      "input_tokens": "number",
      "output_tokens": "number",
    },
  },
  "prompt": "string",
}
`;
